### DISCUSSION TITLE
why opentelmetry do not have agent?

### DISCUSSION THREAD

**Discussion by zdyj3170101136** [open-telemetry/opentelemetry-go#3033] [Q&A]
In jaeger, sdk send trace data to agent, agent send it to collector.
And the agent can run as daemonset, so it can save the connection number with collector.
Why opentelemetry directly send data to collector but not through agent

**Comment by pellared** [comment]
You can configure the Jaeger exporter to send spans to the Jaeger Agent.
See https://github.com/open-telemetry/opentelemetry-go/tree/exporters/otlp/otlptrace/otlptracehttp/v1.8.0/exporters/jaeger#configuration

**Reply by zdyj3170101136** [reply]
I want to know is if the opentelemetry think that remove agent would not have performance issue.
Is too many connection a problem to opentelemetry collector if i use client directly connected to collector?

**Reply by Petrie** [reply]
[Collector](https://opentelemetry.io/docs/collector/) support `Agent` and `Gateway ` mode both.

**Comment by hanyuancheung** [comment]
OpenTelemetry does not remove the **agent** method, you can use the collector to deploy as both agent and gateway. 

As to which way we should consider, it depends on the specific network environment in your production. 

As for me, I prefer to use the collector centrally(as gateway), service could instrument and report data in a lightweight way. Cause our network env is too complicated to deploy other sidecars such as agents.

You could also add my WeChat, so we can analyze and decide which way is better based on your situation.
My WechatID: zhycsuzhang
